tools/diagnostics/base_tech_bundle/__init__.py

```diff
@@ -1,18 +1,98 @@
 from service_bundle import ServiceBundle
 
+import json
+import logging
+import config
+import sdk_cmd
+
+log = logging.getLogger(__name__)
+
 
 class BaseTechBundle(ServiceBundle):
+
+    def __init__(self, package_name, service_name, scheduler_tasks, service, output_directory):
+        self.package_name = package_name
+        self.service_name = service_name
+        self.scheduler_tasks = scheduler_tasks
+        self.service = service
+        self.framework_id = service.get("id")
+        self.output_directory = output_directory
+        self.install_cli()
+
+    @config.retry
+    def install_cli(self):
+        sdk_cmd.run_cli(
+            "package install {} --cli --yes".format(self.package_name),
+            print_output=False,
+            check=True,
+        )
+
+    @config.retry
+    def create_configuration_file(self):
+        rc, stdout, stderr = sdk_cmd.svc_cli(
+            self.package_name, self.service_name, "describe", print_output=False
+        )
+
+        if rc != 0 or stderr:
+            log.error(
+                "Could not get service configuration\nstdout: '%s'\nstderr: '%s'", stdout, stderr
+            )
+        else:
+            self.write_file("service_configuration.json", stdout)
+
+    @config.retry
+    def create_pod_status_file(self):
+        rc, stdout, stderr = sdk_cmd.svc_cli(
+            self.package_name, self.service_name, "pod status --json", print_output=False
+        )
+
+        if rc != 0 or stderr:
```

Comment:
i _think_ there were some cases in past where we can have `rc==0` && `stderr` being non empty. e.g.: CLI prints (or at least used to) prints the warnings (such as version X of CLI is not compatible with version Y of dc/os cluster) to stderr.
May be we can print the `stderr` and write `stdout` to json file if `stdout` is non empty ?!

Reply:
OR we can "try" loading stdout as json and if it succeeds, we can write it to the file.